internal/fullnode/build_pods.go

```diff
@@ -22,12 +22,17 @@ func BuildPods(crd *cosmosv1.CosmosFullNode, cksums ConfigChecksums) ([]diff.Res
 		if err != nil {
 			return nil, err
 		}
-		if disable := overrides[pod.Name].DisableStrategy; disable != nil {
-			continue
-		}
 		if _, shouldSnapshot := candidates[pod.Name]; shouldSnapshot {
 			continue
 		}
+		if o, ok := overrides[pod.Name]; ok {
+			if o.DisableStrategy != nil {
+				continue
+			}
+			if o.Image != "" {
+				pod.Spec.Containers[0].Image = o.Image
+			}
```

Comment:
Using the first image in the slice is the natural logic.  

But I've gotten in trouble with this in the past. The user is able to patch and add whatever containers they want. This possibility is a tradeoff of the operator's flexibility. Therefore, if the user patches, there is no guarantee the `node` container will be the first one. So, I suggest finding the specific node container vs. using the slice index.

I'll push a commit tomorrow morning.